src/web/client/src/components/PasswordReset/PasswordReset.tsx

```diff
@@ -0,0 +1,130 @@
+import React, {
+  Component,
+  FormEvent,
+  ChangeEvent,
+  ComponentState,
+} from "react";
+import Button from "@material-ui/core/Button";
+import TextField from "@material-ui/core/TextField";
+import Grid from "@material-ui/core/Grid";
+import Typography from "@material-ui/core/Typography";
+import Container from "@material-ui/core/Container";
+
+interface Props {
+  match: any;
+  history: any;
+}
+interface State {
+  password: string;
+  successMessage: string;
+  failMessage: string;
+}
+
+export default class PasswordReset extends Component<Props, State> {
+  constructor(props: Props) {
+    super(props);
+    this.state = {
+      password: "",
+      successMessage: "",
+      failMessage: "",
+    };
+  }
+
+  async componentDidMount() {
+    try {
+      const requestOptions = {
+        method: "POST",
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          resetToken: this.props.match.params.token,
+        }),
+      };
+      const res = await fetch("/auth/verifyToken", requestOptions);
+      if (res.status === 400) {
+        this.props.history.push("/");
+      }
+    } catch (e) {
+      console.error(e);
+    }
+  }
+
+  onSubmit = async (event: FormEvent<HTMLFormElement>) => {
+    try {
+      event.preventDefault();
+      const requestOptions = {
+        method: "POST",
+        headers: {
+          Accept: "application/json",
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          password: this.state.password,
+          resetToken: this.props.match.params.token,
+        }),
+      };
+      const res = await fetch("/auth/submitPasswordReset", requestOptions);
+      const msg = await res.json();
+      if (msg === "Successfully updated password") {
```

Comment:
Would it be better to depend on the HTTP status code rather than verifying through string?

Reply:
I use the message because there can be many things that can go wrong with an input. For example, a password is too short or invalid username or password or something. So I can just render out the message to the user. If I were to use status code, I will have to create a state for every scenario that can go wrong. So that's why I decided to depend on the text.

Reply:
ok